### DISCUSSION TITLE
+=类似的赋值运算符需要加吗?

### DISCUSSION THREAD

**Discussion by zhangzhimin** [prajna-lang/prajna#20] [Programming Language Design]
1. 需要将其像C++一样作为一个可定义的运算符吗?

2. 将其单纯等价于 "A += B" -> "A = A + B"的语法糖有必要吗?

**Comment by zhangzhimin** [comment]
目前将其作为"可定义的运算符"肯定是不行的, 因为"+="是"+"的拓展, 定义了"+"就决定了"+="的行为.

我们可以讨论下引入一个语法糖是否必要?

**Comment by microctar** [comment]
这个语法糖有必要，多数现代语言都包含它。
而类似于++和--这样的递增递减运算符是没必要的，需要使用者记住运算顺序且容易引发bug，增加了使用者的心智负担。

**Comment by microctar** [comment]
将其实现为可定义运算符进行实现也需要认真考虑。
在开发中一般是避不开自定义数据类型的，同时也需要自定义运算符。

**Comment by microctar** [comment]
设计好语法糖一来减轻使用者的心智负担，二来改善使用者的开发体验。

**Reply by zhangzhimin** [reply]
主要是对多门语言的开发者不优化, 比如python, 不好切换.很多时候开发者会同时使用多门语言, 使用较小交集容易切换
